### PR TITLE
 Add a way of specifying blockSize and compressLevel to FLAC encod…

### LINKED ISSUE #595: Add a way of specifying frame duration to FLAC encoder config
This is a followup to #405. #551 added a frameDuration attribute to the Opus encoder configuration, and closed #405 since Opus is the only codec with this tuning knob supported _by any browser_.

I'm the author of the polyfill: https://github.com/ennuicastr/libavjs-webcodecs-polyfill . The polyfill does support FLAC encoding.

I'll be adding support for this tuning knob to the Opus encoder in libavjs-webcodecs-polyfill Soon™, but I'd like to also add it to FLAC, and ideally do so following the spec instead of deviating from the spec with the possibility of Chrome later deviating in a different way. That is, I'd like to see similar for FLAC.

### PR DESCRIPTION
**Pull request by bdrtc**
This PR adds blockSize, compressLevel configuration knobs to the FLAC encoder.

This should fix #595 , 
@Yahweasel @dalecurtis

### CODE REVIEW

**Review by dalecurtis — APPROVED** [review]

**Comment by bdrtc** [comment]
@padenot @aboba .

File: flac_codec_registration.src.html [hunk 1]
```diff
@@ -94,6 +102,59 @@
...
+dictionary FlacEncoderConfig {
+  [EnforceRange] unsigned long blockSize = 0;
+  [EnforceRange] unsigned long compressLevel = 5;
+};
+</xmp>
+</pre>
+
+To check if an {{FlacEncoderConfig}} is valid, run these steps:
+1. If {{FlacEncoderConfig/blockSize}} is not a valid block size,
+	which is described section 5.1 of [[FLAC-DRAFT]], return `false`.
```
Comment (chrisn) [inline flac_codec_registration.src.html:138]: Do we need both the `FLAC` and `FLAC-DRAFT` references? The FLAC reference also describes the valid block size. That said, if this is going through the IETF, we should prefer the IETF-DRAFT, then update the registration once it becomes an RFC

**Review by chrisn — CHANGES_REQUESTED** [review]

File: flac_codec_registration.src.html (hunk 1 above)
Comment (bdrtc) [reply flac_codec_registration.src.html:138]: IETF-DRAFT should be more formal, we will record this, and update when it become final RFC.

**Comment by Yahweasel** [comment]
> Do any reviewers have further comments here ? Thanks for your time 1st_place_medal

I have none. Works for me.

**Comment by dalecurtis** [comment]
@chrisn did you want to stamp and merge?

**Comment by chrisn** [comment]
There's just one change remaining, I suggest removing the `FLAC-DRAFT` reference and updating `FLAC` to point to the IETF draft.

**Comment by bdrtc** [comment]
> There's just one change remaining, I suggest removing the `FLAC-DRAFT` reference and updating `FLAC` to point to the IETF draft.

done, update FLAC reference  to  IETF draft.   @chrisn

**Review by chrisn — APPROVED** [review]
Thank you @bdrtc, looks good!

**Pull request merged**